### PR TITLE
NMS-13507: even more speed-up enlinkd link retrieval by bulk fetching

### PR DESCRIPTION
**Pull request by swachter**
Issue: https://issues.opennms.org/browse/NMS-13507

This PR bulk fetches some entities that are used during enlinkd link retrieval.

### CODE REVIEW

**Review by Naicisum — APPROVED** [review]
Tested and functions as expected

File: features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/IsIsElementDao.java [hunk 1]
```diff
@@ -40,6 +42,12 @@
     public IsIsElement findByNodeId(Integer id);
 
     public IsIsElement findByIsIsSysId(String isisSysId);
+
+    /**
+     * Returns all IsIsElements that a have an isisSysID that matches an isisISAdjNeighSysID of an IsIsLink related to the given
```
Comment (christianpape) [inline features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/IsIsElementDao.java:47]: typo: "that a have" -> "that have"

File: features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/OspfElementDao.java [hunk 2]
```diff
@@ -46,6 +46,12 @@
 
     public List<OspfElement> findAllByRouterId(InetAddress routerId);
 
+    /**
+     * Returns all OspfElements that a have an ospfRouterId that matches an ospfRenRouterId of an OspfLink related to the given
```
Comment (christianpape) [inline features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/OspfElementDao.java:50]: Maybe a typo: Isn't it ospfRemRouterId, Rem for remote?

File: opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/SnmpInterfaceDao.java [hunk 3]
```diff
@@ -55,6 +55,12 @@
 
     List<OnmsSnmpInterface> findByMacLinksOfNode(Integer nodeId);
 
+    /**
+     * Returns all OnmsSnmpInterfaces that a have a physAddr that matches an isisISAdjNeighSNPAAddress of an IsIsLink related to the given
```
Comment (christianpape) [inline opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/SnmpInterfaceDao.java:59]: Typo: "that a have" -> "that have"

**Review by christianpape — CHANGES_REQUESTED** [review]
Only typos.

File: features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/OspfElementDao.java (hunk 2 above)
Comment (swachter) [reply features/enlinkd/persistence/api/src/main/java/org/opennms/netmgt/enlinkd/persistence/api/OspfElementDao.java:50]: fixed

**Review by christianpape — APPROVED** [review]
Looks good to me.

**Comment by Naicisum** [comment]
Was this meant to go into `develop` and not `release-29.x`?

**Pull request merged**